synapse/lib/stormlib/gen.py

```diff
@@ -105,10 +105,32 @@ class LibGen(s_stormtypes.Lib):
                        'desc': 'Type normalization will fail silently instead of raising an exception.'},
                   ),
                   'returns': {'type': 'node', 'desc': 'A lang:language node with the given code.'}}},
+        {'name': 'itAvScanResultByTarget',
+         'desc': 'Returns an it:av:scan:result node by deconflicting with a target and signature name, adding the node if it does not exist.',
+         'type': {'type': 'function', '_funcname': '_storm_query',
+                  'args': (
+                      {'name': 'type', 'type': 'str', 'desc': 'The target type.'},
+                      {'name': 'target', 'type': 'str', 'desc': 'The target value.'},
+                      {'name': 'signame', 'type': 'str', 'default': None, 'desc': 'The signature name.'},
+                      {'name': 'scanner', 'type': 'str', 'default': None,
+                       'desc': 'An optional scanner software name to include in deconfliction.'},
+                      {'name': 'time', 'type': 'time',
+                       'desc': 'An optional time when the scan was run to include in the deconfliction.'},
+                      {'name': 'try', 'type': 'boolean', 'default': False,
+                       'desc': 'Type normalization will fail silently instead of raising an exception.'},
+                  ),
+                  'returns': {'type': 'node', 'desc': 'An it:av:scan:result node.'}}},
     )
     _storm_lib_path = ('gen',)
 
     _storm_query = '''
+        function _maybeCast(try, type, valu) {
```

Comment:
Can we mark this as private so we're not exposing it unwittingly. ``__maybeCast``  would do.